### PR TITLE
Add more API features

### LINKED ISSUE #5: [Backend] Add Authorization Rules [4pt]
Add the following authorization rules for the API endpoints.
- All routes can't be accessed without signing in.
- A signed-in user can see all doctors.
- A signed-in user can reserve a doctor.
- An admin can create a doctor.
- An admin can delete a doctor.

### LINKED ISSUE #6: [Backend] API Documentation [3pt]
Write specs for the controllers and generate the API documentation.

### LINKED ISSUE #20: Deploy API [3pt]
Deploy the API to a suitable hosting platform.

### PR DESCRIPTION
**Pull request by preciousuwhubetine**
This PR fixes issues #5, #6, and #20.

Here's a summary of what I did:
- Implementation of authorization using the `cancancan` gem.
- Configuration of CORS.
- Controller testing with RSpec.
- API documentation generation with RSwag.
- API deployment to render.

The API is live at this link: [Link](https://doc-reserve.onrender.com/api/v1).
The API docs can be found [here](https://doc-reserve.onrender.com/api-docs).

### CODE REVIEW

File: app/controllers/api/v1/reservations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :authenticate_user!
+  authorize_resource
+
+  def index
+    @reservations = current_user.reservations.all
+    render json: @reservations, status: :ok
+  end
```
Comment (SabiMantock) [inline app/controllers/api/v1/reservations_controller.rb]: Great work on the reservations _controller! just a quick suggestion to optimize the index action. To address the `N+1` query problem, consider using includes with the Reservation model's association named doctor. This way, you can fetch all reservations for the current user and their associated doctors in a single query, enhancing database performance.
`  @reservations = current_user.reservations.includes(:doctor).all`

**Review by SabiMantock — CHANGES_REQUESTED** [review]
Hello! Thank you for your work on the Auth documentation. I've reviewed the changes, and it's looking good overall. However, there is an important point to address before we merge this.

## User Authentication Documentation: 
It seems that the Swagger documentation is missing information about user authentication. It's crucial to include detailed documentation on how users can authenticate and obtain the necessary API tokens or keys. This will ensure users understand how to securely access the APIs.

Also please check below for comments.

**Comment by preciousuwhubetine** [comment]
@SabiMantock Thanks for the review.
I didn't add documentation for User Authentication because this is not a public API. The API is to be consumed by our own client which will know how to interact with it.
I will add it to the API documentation but public users can't interact with it directly.

I will also work on the reservations controller to avoid N+1 problems. Thanks for pointing that out!

**Review by SabiMantock — APPROVED** [review]
Great job, Approved! go ahead and merge

**Pull request merged**